doc/kernel/other/interrupts.rst

```diff
@@ -206,6 +206,128 @@ The following code demonstrates a direct ISR:
        ...
     }
 
+Implementation Details
+======================
+
+Interrupt tables are set up at build time using some special build tools.
+The details laid out here apply to all architectures except x86, which will
+be covered in its own section.
+
+Any invocation of :c:macro:`IRQ_CONNECT` will declare an instance of
+struct _isr_list which is placed in a special .intList section:
+
+.. code-block:: c
+
+    struct _isr_list {
+        /** IRQ line number */
+        s32_t irq;
+        /** Flags for this IRQ, see ISR_FLAG_* definitions */
+        s32_t flags;
+        /** ISR to call */
+        void *func;
+        /** Parameter for non-direct IRQs */
+        void *param;
+    };
+
+Zephyr is built in two phases; the first phase of the build produces
+zephyr_prebuilt.elf which contains all the entries in the .intList section
+preceded by a header:
+
+.. code-block:: c
+
+    struct {
+        void *spurious_irq_handler;
+        void *sw_irq_handler;
+        u32_t num_isrs;
+        u32_t num_vectors;
+        struct _isr_list isrs[];  <- of size num_isrs
+    };
+
+This is then used by the gen_isr_tables.py script to
+generate a vector table and/or a software ISR table. This is produced as a C
+file and linked into the final application.
+
+The priority level of any interrupt is not encoded in these tables, instead
+:c:macro:`IRQ_CONNECT` also has a runtime component which programs the
+desired priority level of the interrupt to the interrupt controller. Some
+arches do not support the notion of interrupt priority, in which case the
+priority argument is ignored.
+
+Vector Table
+------------
+Whether to generate this depends on whether CONFIG_GEN_IRQ_VECTOR_TABLE is
+enabled. This is simply an array of function pointers, where each element
```

Comment:
A vector table is simply an array of function pointers, where each element **n** corresponds to the IRQ handler for IRQ line **n**, and the function pointers are: